projects/igniteui-angular/migrations/common/UpdateChanges.ts

```diff
@@ -434,6 +453,182 @@ export class UpdateChanges {
+        if (!typeDefs) {
+            // ts property referred in an internal/external template
+            const classDeclaration = langServ
+                .getProgram()
+                .getSourceFile(definition.fileName)
+                .statements.find(m => m.kind === ts.SyntaxKind.ClassDeclaration) as ts.ClassDeclaration;
+            const member: ts.ClassElement = classDeclaration
+                .members
+                .filter(m => m.end === definition.textSpan.start + definition.textSpan.length)[0];
+            if (!member || !member.name) { return null; }
+            typeDefs = langServ.getTypeDefinitionAtPosition(definition.fileName, member.name.getStart() + 1);
+        }
+        if (typeDefs?.length) {
+            return typeDefs[0];
+        }
+
+        return null;
+    }
+
+    private updateClassMembers(entryPath: string, memberChanges: MemberChanges) {
+        let content = this.host.read(entryPath).toString();
+        const changes = new Set<{ change, position }>();
+        for (const change of memberChanges.changes) {
+            const definedIn = change.definedIn.split('|').map(s => s.trim());
+            const matches = this.findMatches(content, change);
+            for (const matchPosition of matches) {
+                const typeDef = this.getTypeDefinitionAtPosition(entryPath, matchPosition - 1);
+                if (!typeDef) { continue; }
+                if (this.memberIsIgniteUI(typeDef.fileName) && definedIn.indexOf(typeDef.name) !== -1) {
+                    changes.add({ change, position: matchPosition });
+                }
+            }
+        }
+
+        const changesArr = Array.from(changes).sort((c, c1) => c.position - c1.position).reverse();
+        for (const fileChange of changesArr) {
+            content = this.replaceMatch(content, fileChange.change.member,
+                fileChange.change.replaceWith, fileChange.position);
+        }
+
+        if (changes.size) {
+            this.host.overwrite(entryPath, content);
+        }
+    }
+
+    private getDefaultProjectForFile(entryPath: string): ts.server.Project {
+        const scriptInfo = this.projectService.getOrCreateScriptInfoForNormalizedPath(ts.server.asNormalizedPath(entryPath), false);
+        this.projectService.openClientFile(scriptInfo.fileName);
+        const project = this.projectService.findProject(scriptInfo.containingProjects[0].projectName);
+        project.addMissingFileRoot(scriptInfo.fileName);
+        return project;
+    }
+
+    private getDefaultLanguageService(entryPath: string): ts.LanguageService | undefined {
+        const project = this.getDefaultProjectForFile(entryPath);
+        return project.getLanguageService();
+    }
+
+    private memberIsIgniteUI(str: string): boolean {
+        return !!this.toRegExp('igniteui-angular').exec(str);
+    }
+
+    private toRegExp(str: string, flags?: string): RegExp {
```

Comment:
So um, there's already an imported `escapeRegExp` method available, so no need for this again? like `new RegExp(escapeRegExp(str), flags);` should do just as well.